---
title: "Analyse de données scRNAseq"
output:
  word_document: default
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown


```{r}
# Seurat v3.0.0 and tximport >=1.13.10
library(Seurat)
library(tximport)

# path to the output directory of Alevin run of pbmc_1k 10x V2 data
alv_quant <- file.path("/home/rstudio/disk/genome/alevin_output/alevin/quants_mat.gz")
file.exists(alv_quant)

# Reading in the alevin quants quants
alv <- tximport(alv_quant, type="alevin")
```

Cell Contrôle qualité
--> On va compter le nombre d'UMI et le nombre de gènes par cellules (pb si trop peu ou trop nombreux)
Low-quality cells or empty droplets will often have very few genes
Cell doublets or multiplets may exhibit an aberrantly high gene count

--> Récupérer les gènes mitochondriaux (si trop nombreux = marquer d'apoptose)
Low-quality / dying cells often exhibit extensive mitochondrial contamination

```{r}
countUMIcell <- apply(alv$counts,2,sum)
hist(countUMIcell, breaks=1000, xlim=range(0,10000))
countgenecell <- colSums(alv$counts !=0)
hist(countgenecell, breaks=7000)

```


```{r}
#on crée la lsit des gènes
listgenes <- rownames(alv$counts)
#on crée la lsite des gènes mitochondriaux
g<-grep("mt-",listgenes)
listmtgenes <-listgenes[g]
listmtgenes
#on crée la matrice des gènes mitochondriaux
mt_counts<-alv$counts[listmtgenes,]
countmtUMI <- apply(mt_counts,2,sum)
hist(countmtUMI, breaks=7000)
```
We calculate mitochondrial QC metrics with the PercentageFeatureSet function, which calculates the percentage of counts originating from a set of features
We use the set of all genes starting with 'mt-' as a set of mitochondrial genes


We filter cells that have unique feature counts over 2,500 or less than 200
We filter cells that have >5% mitochondrial counts

```{r}
library(dplyr)
library(Seurat)
seuratobj <- CreateSeuratObject(counts = alv$counts, project = "TPngs", min.cells = 3, min.features = 200)
seuratobj

# The [[ operator can add columns to object metadata. This is a great place to stash QC stats
seuratobj[["percent.mt"]] <- PercentageFeatureSet(seuratobj, pattern = "mt-")
summary(seuratobj[["percent.mt"]])

# Visualize QC metrics as a violin plot
VlnPlot(seuratobj, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```


